src/Credentials.js

```diff
@@ -312,7 +343,7 @@ async processDisclosurePayload ({doc, payload}) {
   *  @deprecated
   */
   receive (token, callbackUrl = null) {
-    return this.authenticate(token, callbackUrl)
+    return decodeJWT(token).req ? this.authenticate(token, callbackUrl) : this.verifyProfile(token, callbackUrl)
```

Comment:
is breaks the whole point of authenticate. Stick with authenticate here

Reply:
@pelle yeah not the point of the authenticate (but also not authenticate, still receive), but authenticate is the not the same functionality as receive, so it is not backwards compatible if we place authenticate in the receive function. Receive before only verified challenge/req if available. So for example connect always uses receive, even those it doesn't always expect the original request (challenge) to the be the in response (for example when a request is made from provider or unsigned).